### ISSUE TITLE
How it work ?

### ISSUE THREAD

**Issue by luca2125** [dfint/df-translate-hook#1]
Hi this is very interested project !!

How it work ?

Can be used theorically to any games ?

**Comment by aNNiPAk** [comment]
Hey, I haven't had a chance to answer before.

My library is connected to an Exe file. This library intercepts calls to the game's functions during its first download using Detours.
However, to perform such interception you need to use IDA Pro to find the address of such a function and its signature, and then specify it in the library.

Yes, theoretically it can be used in any game. However, for games written using Unity and C#, there are other, lighter options.

**Comment by luca2125** [comment]
Hi Rebert, 

Thank you for your reponse.

If for example I have an extrnal dll with a function (ex Do_ForceFeedback) that accept one paramenter, is possible intercept the offset that I need on a game exe (with IDA Pro or chat engine) and redirect to my function ?

Example:

game exe offset: 04433x  is a collision: call Do_ForceFeedback in my dll with parameter value 1
game exe offset: 53535x  is a damage: call Do_ForceFeedback in my dll with parameter value 2
game exe offset: 6453x4  is a file laser: call Do_ForceFeedback in my dll with parameter value 3
 
In short I like to add the force feedback support on games that not support it.

Do you think is possible do it with your project ?  If not , do you know if exist a project that can do it ?

many thanks !!

**Comment by aNNiPAk** [comment]
Hi, I think so, but we need to convert it according to your needs. To do that, I'd like to look at a concrete example.

**Comment by luca2125** [comment]
Hi Robert,

You have right.

Here there is a concrete example (this project was not mine, but can be useful):

https://github.com/bobhelander/EDForceFeedback

This project add the force feedback support to Elite Dangerous that not support it.

I descrive it in few words:

The main ffb code is:
https://github.com/bobhelander/EDForceFeedback/blob/master/ForceFeedbackSharpDx/MicrosoftSidewinder.cs

where there is a function name "PlayFileEffect" that run the force feedback effect.

This project use a external dll to hook all events of Elite Dangerous:

https://github.com/EliteAPI/EliteAPI

When the a particual event start, run PlayFileEffect that excecute the force feedback effect.

The idea is intercept a useful events of a game (collisions, damage, etc..) and redirect to another function (asyncronally)  in this case "PlayFileEffect" but can be another name.

What you think ?

**Comment by aNNiPAk** [comment]
Hi, I'm sorry about the long answer.

About the help: I must have rushed to reassure you about my skills. It's gonna take me a long time to try and realize what you want. For I don't have the knowledge or skills to do it right now.

My project is mostly based on examples from Youtube, I did not know the hook function before and only interest and a small need pushed me to do so. Try to look for similar examples or go deeper into the topic of hook.
